server/db/dbManager.ts

```diff
@@ -55,14 +111,64 @@ export class UserManager {
 		this.collection = collection;
 	}
 
+	async addPostUpvote(postID: string, username: string): Promise<boolean> {
+		const user = await this.collection.findOne({ username });
+		if (user && !Object.prototype.hasOwnProperty.call(user.upvotes, postID)) {
```

Comment:
The alternative approach to check the property is:
```suggestion
		if (user && !(postID in user.upvotes)) {
```